libkbfs/mdserver_memory.go

```diff
@@ -548,19 +569,54 @@ func (md *MDServerMemory) Put(ctx context.Context, rmds *RootMetadataSigned,
 	return nil
 }
 
+func (md *MDServerMemory) isLockedLocked(ctx context.Context,
+	tlfID tlf.ID, lockID keybase1.LockID) bool {
+	return md.config.Clock().Now().Sub(md.lockIDs[mdLockMemKey{
+		tlfID:  tlfID,
+		lockID: lockID,
+	}]) < mdLockTimeout
+}
+
+func (md *MDServerMemory) lockLocked(ctx context.Context,
+	tlfID tlf.ID, lockID keybase1.LockID) {
+	// NOTE: taking lock would always succeed in MDServerMemory since it's
+	// always within the same session.
+	lockKey := mdLockMemKey{
+		tlfID:  tlfID,
+		lockID: lockID,
+	}
+	existing, ok := md.lockIDs[lockKey]
+	if !ok || md.config.Clock().Now().Sub(existing) >= mdLockTimeout {
+		// The lock doesn't exists or has expired.
+		md.lockIDs[lockKey] = md.config.Clock().Now()
```

Comment:
Why not track the expiration time instead of the lock time?  That would avoid a few of these subtractions.